src/date-time.ts

```diff
@@ -0,0 +1,553 @@
+    */
+    public static createFromValues(dateValue: string, timeValue: string, zone: string): DateTime
+    {
+        given(dateValue, "dateValue").ensureHasValue().ensureIsString()
+            .ensure(t => t.matchesFormat("####-##-##"));
+
+        given(timeValue, "timeValue").ensureHasValue().ensureIsString()
+            .ensure(t => t.matchesFormat("##:##"));
+
+        given(zone, "zone").ensureHasValue().ensureIsString();
+
+        const dateTimeString = `${dateValue} ${timeValue}`;
+
+        return new DateTime({
+            value: dateTimeString,
+            zone
+        });
+    }
+
+    public static min(dateTime1: DateTime, dateTime2: DateTime): DateTime
+    {
+        given(dateTime1, "dateTime1").ensureHasValue().ensureIsType(DateTime);
+        given(dateTime2, "dateTime2").ensureHasValue().ensureIsType(DateTime);
+
+        if (dateTime1.valueOf() < dateTime2.valueOf())
+            return dateTime1;
+
+        return dateTime2;
+    }
+
+    public static max(dateTime1: DateTime, dateTime2: DateTime): DateTime
+    {
+        given(dateTime1, "dateTime1").ensureHasValue().ensureIsType(DateTime);
+        given(dateTime2, "dateTime2").ensureHasValue().ensureIsType(DateTime);
+
+        if (dateTime1.valueOf() > dateTime2.valueOf())
+            return dateTime1;
+
+        return dateTime2;
+    }
+
+    public static validateDateTimeFormat(value: string): boolean
+    {
+        // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+        if (value == null || value.isEmptyOrWhiteSpace())
+            return false;
+
+        return LuxonDateTime.fromFormat(value, DateTime._format).isValid;
+    }
+
+    public static validateDateFormat(value: string): boolean
+    {
+        // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+        if (value == null || value.isEmptyOrWhiteSpace())
+            return false;
+
+        return LuxonDateTime.fromFormat(value, "yyyy-MM-dd").isValid;
+    }
+
+    public static validateTimeFormat(value: string): boolean
+    {
+        // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+        if (value == null || value.isEmptyOrWhiteSpace())
+            return false;
+
+        return LuxonDateTime.fromFormat(value, "HH:mm").isValid;
+    }
+
+    public static validateTimeZone(zone: string): boolean
+    {
+        // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+        if (zone == null || zone.isEmptyOrWhiteSpace())
+            return false;
+
+        try
+        {
+            DateTime._validateZone(zone);
+        }
+        catch
+        {
+            return false;
+        }
+
+        return LuxonDateTime.now().setZone(zone).isValid;
+    }
+
+
+    private static _validateZone(zone: string): void
+    {
+        if (zone.toLowerCase() === "utc")
+            return;
+
+        given(zone, "zone")
+            .ensureWhen(
+                zone.toLowerCase() === "local",
```

Comment:
luxon accepts "local" as a valid zone param and it will take the current system zone as the zone. since it won't be specific. i am just preventing use of local as a valid zone

Reply:
if we want to use local time zone we can use ` DateTime.currentZone`